src/cryptonote_core/blockchain.cpp

```diff
@@ -1125,84 +1125,72 @@ bool Blockchain::validate_miner_transaction(const block& b, size_t cumulative_bl
       return false;
     }
 
-    std::string governance_wallet_address_str;
-    switch (m_nettype)
+    if (b.miner_tx.vout.back().amount != reward_parts.governance)
     {
-      case STAGENET:
-        governance_wallet_address_str = ::config::stagenet::GOVERNANCE_WALLET_ADDRESS;
-        break;
-      case TESTNET:
-        governance_wallet_address_str = ::config::testnet::GOVERNANCE_WALLET_ADDRESS;
-        break;
-      case FAKECHAIN: case MAINNET:
-        governance_wallet_address_str = ::config::GOVERNANCE_WALLET_ADDRESS;
-        break;
-      default:
-        return false;
+      MERROR("Governance reward amount incorrect.  Should be: " << print_money(reward_parts.governance) << ", is: " << print_money(b.miner_tx.vout.back().amount));
+      return false;
     }
 
-    if (!validate_governance_reward_key(m_db->height(), governance_wallet_address_str, b.miner_tx.vout.size() - 1, boost::get<txout_to_key>(b.miner_tx.vout.back().target).key, m_nettype))
+    if (!validate_governance_reward_key(m_db->height(), *cryptonote::get_config(m_nettype, version).GOVERNANCE_WALLET_ADDRESS, b.miner_tx.vout.size() - 1, boost::get<txout_to_key>(b.miner_tx.vout.back().target).key, m_nettype))
     {
       MERROR("Governance reward public key incorrect.");
       return false;
     }
   }
 
+  base_reward = reward_parts.adjusted_base_reward;
   if(base_reward + fee < money_in_use)
   {
-    MERROR_VER("coinbase transaction spend too much money (" << print_money(money_in_use) << "). Block reward is " << print_money(base_reward + fee) << "(" << print_money(base_reward) << "+" << print_money(fee) << ")");
+    MERROR_VER("coinbase transaction spend too much money (" << print_money(money_in_use) << "). Block reward is " << print_money(base_reward) << "(" << print_money(base_reward) << "+" << print_money(fee) << ")");
     return false;
   }
-  // From hard fork 2, we allow a miner to claim less block reward than is allowed, in case a miner wants less dust
-  if (m_hardfork->get_current_version() < 2)
-  {
-    if(base_reward + fee != money_in_use)
-    {
-      MDEBUG("coinbase transaction doesn't use full amount of block reward:  spent: " << money_in_use << ",  block reward " << base_reward + fee << "(" << base_reward << "+" << fee << ")");
-      return false;
-    }
-  }
-  else
-  {
-    // from hard fork 2, since a miner can claim less than the full block reward, we update the base_reward
-    // to show the amount of coins that were actually generated, the remainder will be pushed back for later
-    // emission. This modifies the emission curve very slightly.
-    CHECK_AND_ASSERT_MES(money_in_use - fee <= base_reward, false, "base reward calculation bug");
-    if(base_reward + fee != money_in_use)
-      partial_block_reward = true;
-    base_reward = money_in_use - fee;
-  }
+
+  // since a miner can claim less than the full block reward, we update the base_reward
```

Comment:
honestly since "dust" isn't a thing anymore in this context we may as well just not allow miners to take less of a reward anymore -- that's an artifact of pre-RingCT Monero.

Reply:
We have some core tests atm that still rely on this logic afaik, the ones still relevant for us are namely the bulletproof tests.